toolchain/lex/tokenized_buffer.cpp

```diff
@@ -909,7 +909,8 @@ class [[clang::internal_linkage]] TokenizedBuffer::Lexer {
     // TODO: Need to add support for Unicode lexing.
     if (LLVM_LIKELY(position + 2 >= static_cast<ssize_t>(source_text.size()) ||
                     source_text[position + 1] != '#' ||
-                    !IsIdStartByteTable[source_text[position + 2]])) {
+                    !IsIdStartByteTable[static_cast<unsigned char>(
+                        source_text[position + 2])])) {
```

Comment:
It looks like we have the same error in `LexKeywordOrIdentifier` (L869), and in the macro-generated `Dispatch##LexMethod##SymbolToken` functions (L1075, `OneCharTokenKindTable`), though neither of those can ever actually see a negative `char` value. Would you mind changing those to cast to `unsigned char` too, for consistency?